atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java

```diff
@@ -338,7 +334,7 @@ private void putInternal(TableReference tableRef, Collection<Map.Entry<Cell, Val
                 // Save memory by sharing rows.
                 row = nextKey.row;
             }
-            byte[] oldContents = table.entries.putIfAbsent(new Key(row, col, timestamp), copy(contents));
+            byte[] oldContents = table.entries.putIfAbsent(new Key(row, col, timestamp), Arrays.copyOf(contents, contents.length));
```

Comment:
Sorry, I meant that you should just call your method `copyOf`, the intent to abstract away messing around with lengths was a good one.